MultiSource/Benchmarks/Olden/em3d/make_graph.c

```diff
@@ -123,9 +123,9 @@ void make_neighbors(ptr<node_t> nodelist, array_ptr<table_arr_t> table : count(P
       if ((((unsigned long long) other_node) >> 7) < 2048)
         chatting("post other_node = 0x%x\n",other_node);
 #endif
-      ++other_node->from_count,            /* <----- 12% load miss penalty */
-          other_node->from_values = _Dynamic_bounds_cast<array_ptr<ptr<double>>>(other_node->from_values, count(other_node->from_count)),
-          other_node->coeffs = _Dynamic_bounds_cast<array_ptr<double>>(other_node->coeffs, count(other_node->from_count));
+      other_node->from_values = _Dynamic_bounds_cast<array_ptr<ptr<double>>>(other_node->from_values, count(0)),
+        other_node->coeffs = _Dynamic_bounds_cast<array_ptr<double>>(other_node->coeffs, count(0)),
```

Comment:
Should it not be `count(other_node->from_count + 1)` instead of `count(0)` in both the `_Dynamic_bounds_casts` on lines 126 and 127?

Reply:
If the observed bounds of `other_node->from_values` (or `other_node->coeffs`) uses the value of `other_node->from_count`, then the increment `other_node->from_count++` will result in their observed bounds being `bounds(unknown)`. `other_node->from_count` doesn't have an original value since it has signed integer type.

Reply:
The check `other_node->coeffs = _Dynamic_bounds_cast<array_ptr<double>>(other_node->coeffs, count(0))` does not serve the purpose of making sure that the number of elements in `other_node->coeffs` are in sync with `other_node->from_values`. In other words, the safety that Checked C promises is not really ensured because the dynamic bounds cast with `count(0)` will succeed at runtime if there are zero or more elements in `other_node->coeffs`.

Therefore, I suggest that we change the type of the `from_count` field to unsigned int from int, and also change the `count(0)` in these dynamic bounds casts. Btw, I tried changing the  type on my local machine and the tests are going through fine in the regular LNT run. I am running the benchmarking run. I changed in `MultiSource/Benchmarks/Olden/em3d/em3d.h` on line 42.

In case this does not work, then I suggest that we delete the two dynamic bounds casts and just keep `++other_node->from_values` on this statement. In its current form, this code may mislead a reader or not show Checked C in a good light.

Reply:
Discussed offline - if we can't change the type of `from_count` to unsigned int, then unfortunately I don't think we can remove the dynamic bounds casts. If the type of `from_count` is signed int, then `++other_node->from_count` will result in a compile-time error since the observed bounds of `other_node->from_values` and `other_node->coeffs` will be `bounds(unknown)`.